src/algoliasearch.helper.js

```diff
@@ -119,8 +119,9 @@ var version = require('./version');
  * just an object containing the properties you need from it.
  */
 function AlgoliaSearchHelper(client, index, options) {
-  if (!client.addAlgoliaAgent) console.log('Please upgrade to the newest version of the JS Client.'); // eslint-disable-line
-  else if (!doesClientAgentContainsHelper(client)) client.addAlgoliaAgent('JS Helper ' + version);
+  if (client.addAlgoliaAgent && !doesClientAgentContainsHelper(client)) {
```

Comment:
We don't explicitly test for addAlgoliaAgent usage with a client with  and without the method.

Reply:
I added the tests.

Reply:
What about exposing `addAlgoliaAgent` like we do for `clearCache`, that way we can assume in InstantSearch implementations that the method exists and just call it, even if it doesn't on the client